node_helper.js

```diff
@@ -9,33 +10,34 @@ module.exports = NodeHelper.create({
 
 	async updateWOTrend (city, userAgent) {
 		let url = `https://www.wetteronline.de/wetter/${city}`;
-        let body = await this.getUrl(url, userAgent);
-        let gid = this.findGid(body);
+		let body = await this.getUrl(url, userAgent);
+		let gid = this.findGid(city, body);
 
-        const WO_DAILY_URL = `https://api-app.wetteronline.de/app/weather/forecast?av=2&mv=13&c=d2ViOmFxcnhwWDR3ZWJDSlRuWeb=&location_id=${gid}&timezone=${process.env.TZ}`;
-        let daily_promise = await this.getUrl(WO_DAILY_URL, userAgent);
-        let dailyData = JSON.parse(await daily_promise);
+		if(gid) {
+			const WO_DAILY_URL = `https://api-app.wetteronline.de/app/weather/forecast?av=2&mv=13&c=d2ViOmFxcnhwWDR3ZWJDSlRuWeb=&location_id=${gid}&timezone=${process.env.TZ}`;
+			let daily_promise = await this.getUrl(WO_DAILY_URL, userAgent);
+			let dailyData = JSON.parse(await daily_promise);
 
-        const WO_HOURLY_URL = `https://api-app.wetteronline.de/app/weather/hourcast?av=2&mv=13&c=d2ViOmFxcnhwWDR3ZWJDSlRuWeb=&location_id=${gid}&timezone=${process.env.TZ}`;
-        let hourly_promise = await this.getUrl(WO_HOURLY_URL, userAgent);
-        let hourlyData = JSON.parse(await hourly_promise);
+			const WO_HOURLY_URL = `https://api-app.wetteronline.de/app/weather/hourcast?av=2&mv=13&c=d2ViOmFxcnhwWDR3ZWJDSlRuWeb=&location_id=${gid}&timezone=${process.env.TZ}`;
+			let hourly_promise = await this.getUrl(WO_HOURLY_URL, userAgent);
+			let hourlyData = JSON.parse(await hourly_promise);
 
-		let event = this.extractEvent(dailyData, hourlyData, body);
-		
-		this.getHelper().sendSocketNotification("WETTERONLINE_RESULTS", event);
+			let event = this.extractEvent(dailyData, hourlyData, body);
+			
+			this.getHelper().sendSocketNotification("WETTERONLINE_RESULTS", event);
+		}
 	},
 
-	findGid (body) {
-		const exp = /WO\.geo = (\{[^\}]+\})/m;
-	
-		const match = body.match(exp);
+	findGid (city, body) {
+		const exp = /gid : "([^"]+)"/s;
+		const gid = body.match(exp);
```

Comment:
i don't like the naming here, the "gid" is technically only `gid[1]` (if there's a match at all)

Reply:
Oh, you have a good point. I just changed it.